R/runs.R

```diff
@@ -1,27 +1,52 @@
 #' @include globals.R
 NULL
 
-# Translate metric to value to safe format for REST.
-metric_value_to_rest <- function(value) {
-  if (is.nan(value)) {
-    as.character(NaN)
-  } else if (value == Inf) {
-    "Infinity"
-  } else if (value == -Inf) {
-    "-Infinity"
-  } else {
-    as.character(value)
+#' @importFrom tibble tibble
+#' @importFrom rlang names2
+get_key_value_df <- function(..., .which = -1) {
+  values <- list(...) %>% unlist()
+  keys <- names2(values)
+  args <- as.list(sys.call(which = .which))
+  backup_keys <- args[2:length(args)] %>%
+    as.vector() %>%
+    as.character() %>%
+    make.names()
+  values <- unname(values)
+  for(i in seq_along(values)) {
+    keys[i] <- ifelse(keys[i] == "", backup_keys[i], keys[i])
+  }
+  tibble(
+    key = keys,
+    value = values,
+  )
+}
+
+#' @importFrom checkmate assert
+assert_new_col_length <- function(x, metrics) {
+  nm <- deparse(substitute(x))
+  n_metrics <- nrow(metrics)
+  n_x <- length(x)
+  cnd <- n_x == n_metrics | n_x == 1
+  if(cnd) {
+    return(TRUE)
   }
+  abort(
+    sprintf(
+      'The length of `%s` should be 1 or the same as the number of metrics (%s), not %s.',
+      nm,
+      n_metrics,
+      n_x
+    )
+  )
 }
```

Comment:
I assume this just mimics dplyr?

Reply:
that was the inspiration. i didn't look at how they actually do it tho. the `return(TRUE)` is what `checkmate` does if everything looks good